modules/timepoint_list/php/timepoint_list.class.inc

```diff
@@ -98,6 +98,9 @@ class Timepoint_List extends \NDB_Menu
             foreach ($listOfTimePoints as $currentTimePoint) {
                 $sessionID = new \SessionID($currentTimePoint);
                 $timePoint = \TimePoint::singleton($sessionID);
+                if(!$timePoint->isAccessibleBy($user)) {
```

Comment:
shouldn't `$listOfTimePoints` be filtered before the `if` at line 85?

Reply:
I didn't notice that.. the only thing that `if` statement seems to do is avoid the overhead of `user     = \User::singleton();` which needs to be done to call isAccessibleBy, but good point. I'll clean it up anyways.

Reply:
Okay, I cleaned it up and made `getListOfTimePoints` return an array instead of ?array. (I don't see why we had 2 ways to represent the same thing in the first place, and it's possible that the if empty was there to avoid an error when it returns null..). The diff is a little heavier now, but it's easier to see the actual changes with `?w=1`

Reply:
hopefully,  no one was doing isnull or === null on the returned value of getListOfTimePoints

Reply:
If they were they already had a bug if the function returned `[]` since the function doesn't say what the difference is between null and the empty array..